projects/core/tokens/icon-resolver.ts

```diff
@@ -1,32 +1,94 @@
 import {inject, InjectionToken, type Provider} from '@angular/core';
-import {type TuiStringHandler} from '@taiga-ui/cdk/types';
+import type {TuiHandler, TuiStringHandler} from '@taiga-ui/cdk/types';
 
 import {TUI_ASSETS_PATH} from './assets-path';
 import {TUI_ICON_REGISTRY} from './icons';
 
+const ICON_MODE_PREFIXES = {
+    font: '@font.',
+    image: '@img.',
+    imageMask: '@img.mask.',
+};
+
+type ICON_TARGET_TAG = 'raw' | 'url';
+
+export interface TuiResolvedIcon {
```

Comment:
Let's not change the resolver API. Let's just add `tuiGetIconMode(icon: string): 'img' | 'icon' | 'font'` and use it both in the resolver and in host binding for `data-` type:
```ts
return icon => tuiGetIconMode(icon) === 'font' ? icon.replace('@font.', '') : ...
```